zerver/lib/actions.py

```diff
@@ -2289,6 +2290,28 @@ def check_send_typing_notification(sender: UserProfile, user_ids: List[int], ope
     )
 
 
+def do_send_stream_typing_notification(
+    sender: UserProfile, operator: str, stream_id: int, topic: str
+) -> None:
+
+    sender_dict = {"user_id": sender.id, "email": sender.email}
+
+    event = dict(
+        type="stream_typing",
```

Comment:
I'd like to have this be a `typing` event, not a `stream_typing` event.  To do that, we need to do the following:
* Reorder this branch so that the frontend work to at least make `server_events.js` not crash when seeing a `typing` event with a `stream_id` field in it and no `recipient_user_profiles` field will work.  (It can ignore them in that first commit).  
* Add a new client_capability, called `stream_typing_notifications`, that is plumbed through exactly the same way that `bulk_message_deletion` is.  You can even do this as a new first commit (with it not having a functional effect in that commit, and then the main commit for typing indicators would just do that).
* Then do this backend commit, with this line having `type="typing"`, and logic in `zerver/tornado/event_queue.py` to use a `process_typing_event` little function that only adds the `typing` events that have a `stream_id` field in them if `client.stream_typing_notifications`.  It should let you simplify the API documentation changes as well.

I think with that set of changes, I'd be happy to merge this series through having the complete backend.

Reply:
> logic in `zerver/tornado/event_queue.py` to use a `process_typing_event` little function that only adds the `typing` events that have a `stream_id` field in them if `client.stream_typing_notifications`.
 
 
 We could also tweak `accepts_event` return `False` in case of `stream_id` with `client.stream_typing_notifications` set to `False`. That seems better to me as `process_typing_event` does a bit of duplication from `process_event`. But for now I added `process_typing_event` as you suggested. I'd like to know what you think about the other approach and if there are any issues possible with that approach.

Reply:
Yeah, I think your `accepts_event` strategy is a better way to do this.

Reply:
Ok, done.

* I didn't make any commit for `server_events.js` to not crash when seeing `typing` event with a `stream_id` field but instead set the `client.stream_typing_notifications` to `False` and made it `True` in the frontend commit as the goal was to be able to merge the backend changes without any side effects and making changes to `server_events.js` felt unnecessary since we have `client.stream_typing_notifications`. 
* Added a new client_capability. This is the new first commit.
* Changed `stream_typing` event to `typing`.